models/git/lfs.go

```diff
@@ -319,3 +333,36 @@ func GetRepoLFSSize(ctx context.Context, repoID int64) (int64, error) {
 	}
 	return lfsSize, nil
 }
+
+// IterateLFSMetaObjectsForRepo provides a iterator for LFSMetaObjects per Repo
+func IterateLFSMetaObjectsForRepo(ctx context.Context, repoID int64, f func(context.Context, *LFSMetaObject, int64) error) error {
+	var start int
+	batchSize := setting.Database.IterateBufferSize
+	sess := db.GetEngine(ctx)
+	type CountLFSMetaObject struct {
+		Count int64
+		LFSMetaObject
+	}
+
+	for {
+		beans := make([]*CountLFSMetaObject, 0, batchSize)
+		// SELECT `lfs_meta_object`.*, COUNT(`l1`.id) as `count` FROM lfs_meta_object INNER JOIN lfs_meta_object AS l1 ON l1.oid = lfs_meta_object.oid WHERE lfs_meta_object.repository_id != 4 GROUP BY lfs_meta_object.id
+		sess.Select("`lfs_meta_object`.*, COUNT(`l1`.oid) AS `count`")
+		sess.Join("INNER", "`lfs_meta_object` AS l1", "`lfs_meta_object`.oid = `l1`.oid")
+		sess.Where("repository_id = ? AND created_unix < ?", repoID, time.Now().Add(-24*7*time.Hour))
+		sess.NoAutoTime().GroupBy("`lfs_meta_object`.id")
```

Comment:
`NoAutoTime` is to disable autofill `created` and `updated`, there is no effect here.

Reply:
Yup but GroupBy() is not exposed on the sess object.... I've adjusted things.

I've also adjusted the comment - the `!=` was used by me as an example to make the query in the first place I should have removed it .